pkg/deploy/kubernetes_image_puller.go

```diff
@@ -629,7 +643,8 @@ func UninstallImagePullerOperator(ctx *DeployContext) (bool, error) {
 	}
 
 	// Update CR to remove imagePullerSpec
-	if ctx.CheCluster.Spec.ImagePuller.Enable || ctx.CheCluster.Spec.ImagePuller.Spec != (chev1alpha1.KubernetesImagePullerSpec{}) {
+	if (ctx.CheCluster.Spec.ImagePuller.Enable || ctx.CheCluster.Spec.ImagePuller.Spec != (chev1alpha1.KubernetesImagePullerSpec{})) &&
```

Comment:
Could you use `()` to emphasis the order of operations
`ctx.CheCluster.Spec.ImagePuller.Enable || ctx.CheCluster.Spec.ImagePuller.Spec != (chev1alpha1.KubernetesImagePullerSpec{})`